packages/eas-cli/src/build/utils/resourceClass.ts

```diff
@@ -82,10 +82,21 @@ function resolveAndroidResourceClass(selectedResourceClass?: ResourceClass): Bui
 async function resolveIosResourceClassAsync(
   exp: ExpoConfig,
   projectDir: string,
-  selectedResourceClass?: ResourceClass
+  resourceClassFlag?: ResourceClass,
+  profileResourceClass?: ResourceClass
 ): Promise<BuildResourceClass> {
   const resourceClass =
-    selectedResourceClass ?? (await resolveIosDefaultRequestedResourceClassAsync(exp, projectDir));
+    resourceClassFlag ??
+    profileResourceClass ??
+    (await resolveIosDefaultRequestedResourceClassAsync(exp, projectDir));
+
+  if (resourceClassFlag === ResourceClass.LARGE) {
+    throw new Error(
+      `The experimental "large" resource class for Intel iOS workers is no longer available. Define the avaliable resource class, which you want to use for your builds in eas.json. ${learnMore(
+        'https://docs.expo.dev/build-reference/eas-json/'
+      )}`
+    );
```

Comment:
```suggestion
    throw new Error(
      `Experimental "large" resource class for Intel iOS workers is no longer available. If you want to use a non-default resource class, check the available resource classes at https://docs.expo.dev/build-reference/eas-json/
    );
```

+ use the helper for links.